tools/openshift/ocp-ipi-aws/prep_for_subm.sh

```diff
@@ -1,82 +1,80 @@
 #!/bin/sh
 set -e
+
+# Script args
 OCP_INS_DIR="$(realpath ${1:-.})"
+(( $# == 1 )) || shift && TERRAFORM_ARGS=$@ # e.g. -auto-approve -lock-timeout=3m
```

Comment:
What is this supposed to do? There are a few problems:

* if the number of arguments is 0 (which is allowed, based on the `OCP_INS_DIR` line above), `shift` will run and bail out;
* if the number of arguments is 1, `TERRAFORM_ARGS` will be set to the arguments, *i.e.* the directory.

This will also fail if the user runs `prep_for_subm.sh -auto-approve`, expecting the installation directory to be set to `.` and `TERRAFORM_ARGS` to be set to `-auto-approve`.

Reply:
Regarding `TERRAFORM_ARGS` specifically, you should use an array to preserve the separate arguments:

```suggestion
(( $# == 1 )) || shift && TERRAFORM_ARGS=("$@") # e.g. -auto-approve -lock-timeout=3m
```

Reply:
Thanks @skitt, I fixed it.